apps/cowswap-frontend/src/common/hooks/useOnSelectNetwork.tsx

```diff
@@ -4,31 +4,36 @@ import { getChainInfo } from '@cowprotocol/common-const'
 import { isRejectRequestProviderError } from '@cowprotocol/common-utils'
 import { SupportedChainId } from '@cowprotocol/cow-sdk'
 import { useAddSnackbar } from '@cowprotocol/snackbars'
-import { switchChain } from '@cowprotocol/wallet'
-import { useWeb3React } from '@web3-react/core'
+import { useSwitchNetwork, useWalletInfo } from '@cowprotocol/wallet'
 
 import { useCloseModal } from 'legacy/state/application/hooks'
 import { ApplicationModal } from 'legacy/state/application/reducer'
 
+import { useTradeNavigate } from 'modules/trade/hooks/useTradeNavigate'
+import { getDefaultTradeRawState } from 'modules/trade/types/TradeRawState'
 import { useSetWalletConnectionError } from 'modules/wallet/hooks/useSetWalletConnectionError'
 
 import { useLegacySetChainIdToUrl } from './useLegacySetChainIdToUrl'
 
 export function useOnSelectNetwork(): (chainId: SupportedChainId, skipClose?: boolean) => Promise<void> {
-  const { connector } = useWeb3React()
+  const { active } = useWalletInfo()
+  const switchNetwork = useSwitchNetwork()
+  const tradeNavigate = useTradeNavigate()
   const addSnackbar = useAddSnackbar()
   const closeModal = useCloseModal(ApplicationModal.NETWORK_SELECTOR)
   const setChainIdToUrl = useLegacySetChainIdToUrl()
   const setWalletConnectionError = useSetWalletConnectionError()
 
   return useCallback(
     async (targetChain: SupportedChainId, skipClose?: boolean) => {
-      if (!connector) return
-
       try {
         setWalletConnectionError(undefined)
-        await switchChain(connector, targetChain)
-
+        if (active) {
+          await switchNetwork(targetChain)
+          // When wallet is not connected, then only change the chainId in URL
+        } else {
+          tradeNavigate(targetChain, getDefaultTradeRawState(targetChain))
```

Comment:
See https://github.com/cowprotocol/cowswap/pull/4303/commits/bfa5ac21298d70fda082e37fa73a45788c876fbc

Reply:
Pay attention that `_useWalletInfo()` has `location` in dependencies and uses `getCurrentChainIdFromUrl()` as default value.